cloud/aws/aws_env.h

```diff
@@ -244,6 +251,9 @@ class AwsEnv : public CloudEnvImpl {
   // The S3 client
   std::shared_ptr<AwsS3ClientWrapper> s3client_;
 
+  // Results of last S3 client call; used in stats collection.
+  static thread_local AwsS3ClientResult s3client_result_;
```

Comment:
Unfortunately we cannot use thread_local, since RocksDB can run on platforms that don't support it. We can use `__thread`, but only in some scenarios, check out how RocksDB does it: https://github.com/facebook/rocksdb/blob/master/monitoring/perf_context.cc#L20